tests/lib/fakeinstaller/main.go

```diff
@@ -0,0 +1,232 @@
+	}
+
+	iconst := &install.Constraints{AllPartitions: true}
+	created, err := install.CreateMissingPartitions(gadgetRoot, diskLayout, lvol, iconst)
+	if err != nil {
+		return fmt.Errorf("cannot create parititons: %v", err)
+	}
+	logger.Noticef("created %v partitions", created)
+
+	return nil
+}
+
+func runMntFor(label string) string {
+	return filepath.Join(dirs.GlobalRootDir, "/run/mnt/", label)
+}
+
+func postSystemsInstallSetupStorageEncryption(details *client.SystemDetails) error {
+	// TODO: check details.StorageEncryption and call POST
+	// /systems/<seed-label> with "action":"install" and
+	// "step":"setup-storage-encryption"
+	return nil
+}
+
+func postSystemsInstallFinish(details *client.SystemDetails) error {
+	// TODO: call POST /systems/<seed-label> with
+	// "action":"install" and "step":"finish"
+	return nil
+}
+
+// XXX: pass in created partitions instead?
+// XXX2: should we mount or will snapd mount?
+func createAndMountFilesystems(bootDevice string, volumes map[string]*gadget.Volume) error {
+	// only support a single volume for now
+	if len(volumes) != 1 {
+		return fmt.Errorf("got unexpected number of volumes %v", len(volumes))
+	}
+
+	disk, err := disks.DiskFromDeviceName(bootDevice)
+	if err != nil {
+		return err
+	}
+	vol := firstVol(volumes)
+
+	for _, stru := range vol.Structure {
+		if stru.Label == "" || stru.Filesystem == "" {
+			continue
+		}
+
+		part, err := disk.FindMatchingPartitionWithPartLabel(stru.Label)
+		if err != nil {
+			return err
+		}
+		// XXX: reuse
+		// gadget/install/content.go:mountFilesystem() instead
+		// (it will also call udevadm)
+		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+			return err
+		}
+
+		// mount
+		mountPoint := runMntFor(stru.Label)
+		if err := os.MkdirAll(mountPoint, 0755); err != nil {
+			return err
+		}
+		// XXX: is there a better way?
+		if output, err := exec.Command("mount", part.KernelDeviceNode, mountPoint).CombinedOutput(); err != nil {
+			return osutil.OutputErr(output, err)
+		}
+	}
+	return nil
+}
+
+func createClassicRootfsIfNeeded(rootfsCreator string) error {
+	dst := runMntFor("ubuntu-data")
+
+	if output, err := exec.Command(rootfsCreator, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+func createSeedOnTarget(bootDevice, seedLabel string) error {
+	// XXX: too naive?
+	dataMnt := runMntFor("ubuntu-data")
+	src := dirs.SnapSeedDir
+	dst := dirs.SnapSeedDirUnder(dataMnt)
+	if err := os.MkdirAll(filepath.Dir(dst), 0755); err != nil {
+		return err
+	}
+	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+// XXX: or will POST {"action":"install","step":"finalize"} do that?
+func writeModeenvOnTarget(seedLabel string) error {
+	// TODO: write modeenv
+	return nil
+}
+
+func run(seedLabel, bootDevice, rootfsCreator string) error {
+	if len(os.Args) != 4 {
```

Comment:
Maybe this check should be in main()? We are already accessing some argument there.